com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/flowable/FlowableFacade.java

```diff
@@ -247,13 +249,19 @@ public void trigger(String userId, String executionId) {
             processEngine.getIdentityService()
                 .setAuthenticatedUserId(userId);
             processEngine.getRuntimeService()
-                .trigger(executionId);
+                .trigger(executionId, getVariablesWithUser(userId));
```

Comment:
Several things:
1. The issue is partially solved. trigger() is not the only method which should pass this variable. Consider deleteProcessInstance() (which is called when doing abort), also moveDeadLetterJobsToExecutableJobs() (which is called on retry), check all usages and paths for this issue to occur and prevent it (including internal slp implementation). Some of these places will also lead to the need for this variable in order to use the Client and the issue will persist for the other scenarios.

2. FlowableFacade should not be aware of the presence of specific variables (in this case VAR_USER), it is a facade and should be abstract. This is a responsibility of the Actions (resume, abort, retry etc.) in this case. Could you move the additional logic in this class where it belongs - actions?

Be aware that the Actions are also in cf.core which is semantically incorrect. They are directly connected to cf.process, as they heavily manage our processes. It is better to move them to cf.process, this way you'll also be able to use Constants.VAR_USER from there instead of defining the variable above (which is code duplication).

Reply:
Addition to the Valio's comment - Is there a reason why passing this userId as variable and not use the Authentication service as before ? Check my below comment